### PR TITLE
Fixed n_pc bug in generator.py

### PR DESCRIPTION
**Pull request by johnjasa**
## Purpose
I realized that we can't change `n_pc` without error due to a small bug in `generator.py`.
Fixed that bug and added a test that uses the 15MW turbine and changes `n_pc` away from the default so we have something testing that functionality.
That test should be useful for other times we want to check the top-to-bottom propagation of options in the yaml files.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
Changing `n_pc` in `modeling_options.yaml` used to not work; now it does. Test added through the form of a reference turbine example.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run existing tests which pass locally with my changes
- [x] I have added new tests or examples that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by gbarter — APPROVED** [review]
I see the n_pc bug, but I'm not following the background story on the changes.  There are a lot of RotorSE constants that would be good to test non-default values.  Not sure we need separate modeling yaml files for all of them, but everything seems okay here otherwise.

**Comment by johnjasa** [comment]
I don't think I explained this well, let me try again. This singular test I added could contain and test the modified yaml options. We wouldn't need separate yamls for each option, just one test where we could change as many parameters off the defaults as we want.

Specifically, no test was doing that for `n_pc` until this one, and I could imagine there are other options that aren't being tested off-default as well. When we find those, we can add them to this existing test.

**Comment by gbarter** [comment]
Okay, I am on board with that.  Maybe a slight desire to keep that sort of test-yaml in the test directory and out of the examples though so as not to clutter or confuse if possible.

**Comment by johnjasa** [comment]
Right, I think the current test script and modified modeling_options should be in the test folder right now, which I think satisfies your suggestion. Is that correct or would you like to see the file paths changed?

**Comment by gbarter** [comment]
Sorry about that.  All is good.  I think I confused myself early in the morning thinking this was all happening in the examples folder.
